pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -1098,71 +1098,19 @@ func (c *Cloud) ensureLoadBalancer(namespacedName types.NamespacedName, loadBala
 		}
 
 		{
-			// Sync listeners
-			listenerDescriptions := loadBalancer.ListenerDescriptions
-
-			foundSet := make(map[int]bool)
-			removals := []*int64{}
-			for _, listenerDescription := range listenerDescriptions {
-				actual := listenerDescription.Listener
-				if actual == nil {
-					klog.Warning("Ignoring empty listener in AWS loadbalancer: ", loadBalancerName)
-					continue
-				}
+			createRequest, deleteRequest := syncElbListeners(loadBalancerName, listeners, loadBalancer.ListenerDescriptions)
```

Comment:
Because this method is big and complicated, I would prefer that `syncElbListeners` only returns the addition/deletion listeners (and leave request type creation as part of ensureLoadBalancer) to reduce the amount of changes here. What do you think?

Reply:
sure - I guess that part doesn't matter that much to me.